### ISSUE TITLE
Provide External Configurations of Scoop URLs and Buckets for On-Premise Deployments

### ISSUE THREAD

**Issue by damnhandy** [ScoopInstaller/Scoop#668]
I'm in the process of setting up an internal deployment of Scoop for use by our development teams. Because my organization wants strict control over the what tools are available, our buckets contain a subset of what is available in standard Scoop. Additionally, we offer additional configurations for use in our environment that we'd not want to make publicly available. Ideally, I'd like to stay in sync with the upstream code with my internal deployment, but there's a few gotchas such as:
- URLs in [`bin\install.ps1`](https://github.com/lukesampson/scoop/blob/master/bin/install.ps1#L8) must be changed, resulting in a completely forked file. This is might be the nature of the beast.
- [`libexec\scoop-update.ps1`](https://github.com/lukesampson/scoop/blob/master/libexec/scoop-update.ps1#L43) is hard coded to use http://github.com/lukesampson/scoop. This took a while to figure out why my main bucket started pull files directly from the public internet.
- Perhaps offer a way to disable the main bucket so that it always defaults to the internal bucket(s)

For the most part, everything is working fine for me right now, but it'd be much easier if we consider externalizing the URIs from the installer script. I know there's the `.scoop` config file, but I haven't had much success in getting it work against a custom repository. I

**Comment by deevus** [comment]
I believe if you set `SCOOP_BRANCH` and `SCOOP_REPO` in `.scoop` after you install but before you do your first update, it will happily run from a custom repository. 

Alternatively, if you delete the `.git` directory from `apps\scoop\current`, set the correct `.scoop` values as above, then update: you should get the same result.

**Comment by damnhandy** [comment]
Let me give that a shot, that might be the trick I'm looking for.

**Comment by damnhandy** [comment]
Okay, this now works. I've been able to fork the installer script and run the scoop config commands to set the correct repo and branch names. With that said, I would close this issue and I will open other issues for more specific changes or additions to the scoop config file.

**Comment by rivy** [comment]
@damnhandy , you might want to look at my fork (https://github.com/rivy/scoop). I encountered this same issue while maintaining my patches on top of the baseline scoop, and, so, refactored all the install and update code to correctly pull from the installation branch. There is no reliance on a local config file. It makes maintaining an independent fork much easier.

Installation and updates expect to be from either 'master' or trial ('tr-*') release branches. The README, bin\install.ps1, and lib\core.ps1 must all maintain correlation, but I implemented tests to make sure that condition isn't violated.

Let me know if that helps, or if you have any questions.
